src/Uno.UI.Composition/Composition/CompositionVisualSurface.skia.cs

```diff
@@ -0,0 +1,73 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.Intrinsics.Arm;
+using System.Text;
+using System.Threading.Tasks;
+using SkiaSharp;
+using Uno.UI.Composition;
+using Windows.Graphics.Display;
+
+namespace Windows.UI.Composition
+{
+	public partial class CompositionVisualSurface : CompositionObject, ICompositionSurface, ISkiaSurface
+	{
+		private SKSurface? _surface;
+		private DrawingSession? _drawingSession;
+
+		SKSurface? ISkiaSurface.Surface { get => _surface; }
+
+		void ISkiaSurface.UpdateSurface(bool recreateSurface)
+		{
+			if (_surface is null || _drawingSession is null || recreateSurface)
+			{
+				_drawingSession?.Dispose();
+				_surface?.Dispose();
+
+				var size = SourceSize != default ? SourceSize : (SourceVisual is not null && SourceVisual.Size != default ? SourceVisual.Size : new Vector2(1000, 1000));
```

Comment:
it's a big enough value for the initial frame, it will automatically change once the visual size gets determined

Reply:
Would PositiveInfinity work and be better choice to express the intent?

Reply:
iirc using PositiveInfinity (or very high values) makes Skia return null when creating the surface

Reply:
looks like Windows uses 100,000 (I assume this is the maximum possible value it supports?) when it can't figure out the size, but that is very much and would allocate much vram
![image](https://github.com/unoplatform/uno/assets/34550324/b4995ef3-5c25-49d8-b077-19b9009af4d7)